src/partition/repart.c

```diff
@@ -3640,6 +3640,9 @@ static int partition_target_sync(Context *context, Partition *p, PartitionTarget
         } else if (t->fd >= 0) {
                 struct stat st;
 
+                if (fsync(t->fd) < 0)
+                        return log_error_errno(errno, "Failed to sync changes to '%s': %m", t->path);
```

Comment:
If it's "just in case", shouldn't the error be logged and ignored?

Reply:
I don't think so, if this fails there's definitely something horribly wrong. The "just in case" is more saying that we already expect this to be synced, but you never know, so let's sync it again just to be sure.